### PR TITLE
Rewriting main.js and settings.js

### LINKED ISSUE #25: Upgrade electron-settings to v3
There are quite a bit of changes that have been introduced with the update to this library. The main change being the settings are no longer stored asynchronously. Second of which there is no default function and the changes to the library make it rather easy to set default settings - which was an issue working with #14. 

Changes:
https://github.com/nathanbuchar/electron-settings/wiki/v3-Release

### PR DESCRIPTION
**Pull request by zosman1**
Including upgrade to electron-settings v3
complete rewrite of main.js and settings.js

Closes #25

### CODE REVIEW

**Comment by dhaynespls** [comment]
reviewed, it looks good to me

File: app/main.js [hunk 1]
```diff
@@ -1,109 +1,110 @@
...
-					console.log(`down function completed succesfully on file: ${file}`);
-					notifyUser("The File Have Succesfully Been Moved Down To The Local Machine!", "grey", 7000);
-				}
+	if (ready){
+		if (filelocation == "down") {
+			notifyUser("The Virtual Machine is currently down on the local machine!", "lightblue", 3000);
+		} else {
+			filelocation = "down";
+			let ifError = false;
+			files.forEach((file) => {
```
Comment (tosman) [inline app/main.js]: Looks like up and down do the same thing minus some values, can we abstract this into a method and use it for both?

File: app/index.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 const electron = require("electron");
 const path = require("path");
+const ipc = require("electron").ipcMain;
```
Comment (akshaykarthik) [inline app/index.js]: why not just `electron.ipcMain`? why re-require, in most implementations the require will be cached so it doesn't matter but either way

File: app/main.js [hunk 3]
```diff
@@ -1,109 +1,80 @@
 const fs = require("fs-extra");
 const settings = require("electron-settings");
+const path = require("path");
+const ipc = require("electron").ipcRenderer;
 
-let lastDone = "up"; // imporant var used around main.js
+let ready = false;
+let filelocation = "up";
+const debug = true;
```
Comment (akshaykarthik) [inline app/main.js]: while this works for now, it's usually convention in the node world to use the `NODE_ENV` or `NODE_DEBUG` variables to toggle debug behavior

File: app/main.js [hunk 4]
```diff
@@ -1,109 +1,80 @@
...
+	runningNotifications.push([content, color]);
+	setTimeout(function() {
+		runningNotifications.shift();
+		updateDisplay();
+	}, time);
+	updateDisplay();
 }
-
-exports.notifyUser = function nU(content, color, time) {
+module.exports.notifyUser = function nU(content, color, time){
```
Comment (akshaykarthik) [inline app/main.js]: you don't need to create a separate function to expose this, you can do 
``` module.exports.notifyUser = function notifyUser(content, color, time) {
 	runningNotifications.push([content, color]);
	setTimeout(function() {
 		runningNotifications.shift();
 		updateDisplay();
 	}, time);
 	updateDisplay();
};```

File: app/index.js [hunk 5]
```diff
@@ -1,7 +1,10 @@
 const electron = require("electron");
 const path = require("path");
+const dialog = require("electron").dialog;
```
Comment (akshaykarthik) [inline app/index.js]: same thing here, dialog can bn `electron.dialog`

Comment (zosman1) [inline app/index.js]: Resolved:
https://github.com/zosman1/The_Archiver/pull/27/commits/f1c6627631135fd67e6dc97bfb35111c0b064973

File: app/main.js [hunk 6]
```diff
@@ -1,109 +1,80 @@
...
-			});
+ // eslint-disable-next-line no-unused-vars
+function moveFiles(direction){
+	let fileArgs = {
+		up: ["path.home", "path.away"],
+		down: ["path.away", "path.home"]
+	};
+	if (!ready) return;
+	if (direction != filelocation){
+		filelocation = direction;
```
Comment (zosman1) [inline app/main.js]: resolved
https://github.com/zosman1/The_Archiver/pull/27/commits/0be33ec7d61ae430bc1afcd73f6340b931b83c4e

**Pull request merged**